src/engine/SpatialJoin.h

```diff
@@ -5,15 +5,26 @@
 
 #pragma once
 
+#include <optional>
+
 #include "engine/Operation.h"
+#include "global/Id.h"
 #include "parser/ParsedQuery.h"
 
+// Configuration to restrict the results provided by the SpatialJoin
+struct NearestNeighborsConfig {
+  long long maxResults_;
+  std::optional<long long> maxDist_ = std::nullopt;
+};
+struct MaxDistanceConfig {
+  long long maxDist_;
```

Comment:
See above, the maxResults should probably be `size_t`. And if you want, you can refactor everything s.t. the distance always is a double, but that probably breaks a lot of the current tests, so you can probably leave this for now for the sake of merging this sooner.